### ISSUE TITLE
When moving through chords in piano roll, only play note preview if cursor position changes.

### ISSUE THREAD

**Issue by ScottChesworth** [jcsteh/osara#830]
@ptorpey asked on RWP whether we can clarify the UX around reaching the end of navigable notes in the piano roll. This would also apply to the start when navigating to previous chord.
Our reporting style is consistent already, if the cursor position doesn't change, OSARA doesn't report. However, note previews do play irrespective of whether the cursor position has actually changed. Navigating chords with our new config options to slim down the amount of reporting exasipates this inconsistency.

Do we have enough control over when note previews play to bring those in line with our reporting style? Ideally we'd only hear a preview when navigating to previous or next chord results in the cursor position changing.

**Comment by RichCaloggero** [comment]
I like it. Sounds clean and consistent.

**Comment by jcsteh** [comment]
> Our reporting style is consistent already, if the cursor position doesn't change, OSARA doesn't report.

It's not quite this simple. That is only true (for both the MIDI editor and elsewhere) if report full time is disabled. If report full time is enabled, you'll get the start or end repeated if you keep tapping. In other words, the fact that we don't repeatedly report the same time is part of the partial time caching.

That makes the UX here a bit interesting. If we implement this, if you have report full time enabled, you won't hear the preview repeated on each tap, but you will hear the time. It feels wrong to make preview repeats follow the report full time setting and this would be inconsistent with scrubbing.

To make things more complicated, there's also the question of what we do regarding note count when report full time movement is enabled. If we don't play the preview, should we also not report the note count?

**Comment by ptorpey** [comment]
If report full time is enabled, you'll get the start or end repeated if you keep tapping. 

[PT] For some this might be considered a feature since it makes it obvious when one has come to the end of an item and can’t navigate past it.  If one only hears the note each time one tries to navigate past the beginning or end of an item, one doesn’t really know if there are repeated notes or one is at the end.

[PT] 

--Pete

**Comment by ScottChesworth** [comment]
I just spent a few minutes editing MIDI with report full time enabled in a project that has a bunch of short items. Two observations spring to mind:
1. With that configuration, I think the note preview not playing if the cursor doesn't move would be a valuable clue that you've reached the beginning or end of the item. As things stand I couldn't find a quick way to ascertain when I'd reached the first or last notes of an item without attempting to move multiple times, listening through to a minimum of two repeated reports to compare the percentages. Some sort of more immediate clue would've been very welcome here.
2. I get the importance of making sure every additional option we add serves a clear purpose, but I'm not sure we can make every potential combination of those options make sense 100% of the time. Report full time being un-checked is far and away the more widely adopted config and I'm not seeing any signs of that shifting, so IMO it makes sense to assume that'll likely be the config that's being used. Of course, we keep an eye open to make sure we're not completely hosing any of the alternatives that could be in play, but really, I think you'd be hard pressed to find more than a handful of people who'll choose to edit MIDI with report full time checked. Based on what I've experienced tonight, if I was editing on any sort of time crunch, it would've been agony.

> there's also the question of what we do regarding note count when report full time movement is enabled. If we don't play the preview, should we also not report the note count?
> 
Not reporting note count unless we've actually moved to a different note makes sense to me. If nothing's changed when I've pressed an arrow, I think silence is a clear and intuitive indication that the movement I attempted to make didn't happen as expected.
